src/machine/machine_stm32_i2c.go

```diff
@@ -464,13 +436,7 @@ func (i2c I2C) controllerRequestRead(addr addressable, option transferOption) er
 	}
 
 	// send peripheral address
-	switch addr.bitSize() {
-	case 7: // 7-bit peripheral address
-		i2c.Bus.DR.Set(addr.toRead())
-
-	case 10: // 10-bit peripheral address
-		// TODO
-	}
+	i2c.Bus.DR.Set(directionRead.shift7Bit(addr))
```

Comment:
Same here:
```suggestion
	i2c.Bus.DR.Set(uint32(addr)<<1 | 1)
```

Reply:
Replaced with inline shift, removed type and constants for `i2cDirection`